lib/litani_report.py

```diff
@@ -93,8 +99,9 @@ def render_to_file(self, out_file):
         return not proc.returncode
 
 
-    @staticmethod
-    def render(render_root, pipe_url, pipe):
+    def render(self, render_root, pipe_url, pipe):
+        if not self.should_render:
+            return
         pdr = PipelineDepgraphRenderer(pipe)
```

Comment:
Why construct a new `PipelineDepgraphRenderer` here? `render` is an instance method now. It should just call `self.render_to_file` instead of `pdf.render_to_file`. If you remove the `pipe` instance variable, you could add a `pipe` argument to `render_to_file` and pass it in.